staff/acct/sorry/sorry

```diff
@@ -1,156 +1,202 @@
+        log(rootstaffer, username, filep.read())
+        user_mail = ocflib.account.utils.get_email(username)
+        if user_mail:
+            send_sorry_mail(user_mail, username, filep.read())
+        choice = input("Do you want this script to run 'note -u {0} reason' for you? [Y/n]".format(username))
+        if choice not in ['n', 'N']:
+            subprocess.run(['sudo', '-u', rootstaffer, 'note', '-u', format(username), filep.read()])
+        else:
+            print('NOW RUN note -u {0} reason. NOW'.format(username))
+
+    if kerberos_new_ticket:
+        kerberos_destroy()
+
+
+# Returns the real user who is running this script, but probably only works on Linux
+def real_user():
+    if 'SUDO_USER' in os.environ and os.environ['SUDO_USER'] != 'root':
+        return os.environ['SUDO_USER']
+    else:
+        print("The sorry.log is much more useful when it logs who you are rather than simply 'root'.")
+        username = input('Please enter your username: ')
+        return username
+
+# Initiates a kerberos ticket for current user
+# Returns True when a new ticket is created via kinit so that we destroy it afterwards
+# Raises OSError is kinit screwed up
+
+
+def kerberos_init(staff_name):
+    if ('SORRY_KRB5CCNAME' in os.environ):
+        print('SORRY_KRB5CCNAME set in environment. Assuming this file contains current admin credentials.')
+        os.environ['KRB5CCNAME'] = os.environ['SORRY_KRB5CCNAME']
+        return False
+    else:
+        print('You are {0}.'.format(staff_name))
+        klist = subprocess.run(['sudo', '-u', staff_name, 'klist', '--json'], stdout=subprocess.PIPE)
+        if klist.returncode == 0:
+            cache_info = json.loads(klist.stdout.decode())
+            if cache_info.get('principal') == '{0}/admin@OCF.BERKELEY.EDU'.format(staff_name):
+                print('Using your preexisting credentials.')
+                return False
+
+        if (subprocess.call(['kinit', '{0}/admin'.format(staff_name)]) != 0):
+            raise OSError('Kinit failed, bailing out!')
+        else:
+            print('Kinit complete.')
+            return True
+
+
+def kerberos_destroy():
+    return subprocess.call(['kdestroy'])
+
+# Do all kinds of logging, like sending e-mails or writing to files
+# Change here if more crazy thingy need to be added
+
+
+def log(root_staffer, sorried_user, reason):
+    # Send log files only for now
+    send_log_mail(root_staffer, sorried_user, reason)
+
+# TODO: Do we want a log file or log email?
+
+
+def write_log_file(root_staffer, sorried_user):
+    #
+    with open(SORRY_LOG, 'a') as log_file:
+        # Date style matters...?
+        log_file.write('{0} - {1} {2}'.format(str(datetime.datetime.now()), root_staffer, sorried_user))
+
+
+def send_log_mail(root_staffer, sorried_user, reason):
+    # Use root@ for now
+    email_from = 'Open Computing Facility <root@ocf.berkeley.edu>'
+    email_subject = '[OCF] Account disabled'
+    email_body = textwrap.dedent('''
```

Comment:
Nit here and throughout: per PEP8, triple quotes should always be `"""`, not `'''`.